ports/raspberrypi/common-hal/socketpool/Socket.c

```diff
@@ -148,8 +148,10 @@ STATIC void lwip_socket_free_incoming(socketpool_socket_obj_t *socket) {
         for (uint8_t i = 0; i < alloc; ++i) {
             // Deregister callback and abort
             if (tcp_array[i] != NULL) {
+                MICROPY_PY_LWIP_ENTER;
```

Comment:
Most of this file is copied from micropython's implementation (in their source tree at extmod/modlwip.c)

I notice that this one is not present in upstream micropython. We should raise an issue with them. It could clarify whether this is not needed, or whether they have a bug that would be fixed by this.

I did a quick analysis and I think this addition is probably not necessary, but it could also be that I'm fuzzy on the rules and requirements for using these ENTER/EXIT macros.

This function is called from two contexts: from the _lwip_tcp_error callback and from the ioctl function.

The ioctl function seems to already hold the lock at the site of the call.

Is the lock needed in callback contexts? Not that I did any heavy analysis, but I thought it wasn't.

This general question about the need to ENTER/EXIT within callback functions applies at several places below, I won't call them out individually.

Reply:
You are probably right. I didn't check the context functions are called from. I just put it everywhere just to be safe.